04_fep/05_ATM/01-ATM.ipynb

```diff
@@ -0,0 +1,481 @@
+    "\n",
+    "In this section we will cover a full minimisation and equilibration protocol for the ATM system we created above. We won't actually run any simulations as they would take far too long, but we will still cover each unique step in the protocol.\n",
+    "\n",
+    "ATM simulations are best run with a series of restraints:\n",
+    "- __core alignment__; these are the the rigid core restraints that are applied to the atoms we found earlier.\n",
+    "- __positional restraints__; these are flat-bottom restraints used to keep specific atoms in place, these are generally applied to the alpha carbons of the protein.\n",
+    "- __centre of mass distance restraints__; these are used to maintain the relative positions of the protein and ligands and are applied to the centre off mass of the atoms listed in the `data` dictionary (these were found automatically when we prepared the system, but they can be set manually if the ones found are not appropriate).\n",
+    "\n",
+    "Now let's find the alpha carbons that we will be restraining using BioSimSpace search functionality."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8727fdbb-305f-459a-b8c6-eb5a77424b36",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ca = [atom.index() for atom in solvated.search(\"atomname CA\")]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4ab9036b-5cd8-40ee-b1e4-38dff16eb9d4",
+   "metadata": {},
+   "source": [
+    "Now we can create a minimisation protocol using the forces and `atm_data` dictionary we created when setting up the system."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e1de322e-f178-444e-bfc7-8f5936182b96",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "minimisation = BSS.Protocol.ATMMinimisation(\n",
+    "    data=atm_data, core_alignment=True, restraint=ca, com_distance_restraint=True\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "56b04fa5-aa1e-4032-9ca6-e79a21240845",
+   "metadata": {},
+   "source": [
+    "This protocol can then be run using the BioSimSpace openMM runner."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "03f63303-98be-48b0-ba1a-3c82653a4c30",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "minimisation_process = BSS.Process.OpenMM(solvated, minimisation)\n",
+    "minimisation_process.start()\n",
+    "minimisation_process.wait()\n",
+    "minimised = minimisation_process.getSystem(block=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "31954b81-0307-42d3-8c4c-489a62be74c1",
+   "metadata": {},
+   "source": [
+    "We will now run our first equilibration, including all of the forces we used in the minimisation. "
```

Comment:
test if it is possible to reduce runtime for the tutorial to ~ 10 ps. If so explicitly mention in the comment cell that the values used here are lower than what would be used in a production setting to allow (near) interactive use of the notebook

Reply:
I've done some testing both locally and on the server and it seems that this system is a lot more forgiving than some of the others I've tested, so we can get away with running for only ~1ps during each equilibration step. The issue is that the system is quite large, so even with these small runtimes the total minimisation/equilibration pipeline takes around 10 minutes.

I had originally designed the notebook to have all of the code that actually runs the simulation commented out, but somewhere along the line I must've uncommented them for testing and forgotten about it. This is why the notebook loads a pre-prepared system at the end of the pipeline, since it wasn't actually meant to run any simulations.

I can't guarantee that production simulations run with the output of such a short min/eq pipeline will be stable, so i think the `ready_for_production.bss` file will still be useful as this is from a fully prepared system that has been equilibrated for a total of 700 ps.

I'll reduce the runtimes in the notebook and add a note at the end that clarifies that, in a real production pipeline, users would need to run much longer minimisations and equilibrations in order to ensure stability.

Reply:
The comment cell between equilibration and annealing is now longer in sync with the code 

In this case we will use 10 annealing cycles with a total runtime of 100ps, meaning that the value of lambda is increased by a value of 0.05 every 10ps.
-->
In this case we will use 10 annealing cycles with a total runtime of 1ps, meaning that the value of lambda is increased by a value of 0.05 every 0.1ps. For production runs we recommend extending the runtime to at least 100 ps.